### PR TITLE
add eig

### LINKED ISSUE #14: Eigenvalues and eigendirections for tensors
Define eigenvalues and eigendirections for 2nd order tensors to compute for instance principal stresses and directions. So that

``` jl
σ = rand(SymmetricTensor{2,3})
Λ, Φ = eig(σ)
```

works. I guess `Φ` can be returned as first order tensors, and `Λ` a regular array the way that `eig` works for matrices. But it could also be neat if it returned second order tensors so that you can get back `σ` as

``` jl
σ = Φ*Λ*Φ'
```

which works for symmetric tensors at least. Then `Λ` would be a diagonal second order tensor and  `Φ` a second order tensors as well. The problem is that `σ` would not be a `SymmetricTensor{2,3}` but a `Tensor{2,3}`.

``` jl
julia> using ContMechTensors

julia> σ = rand(SymmetricTensor{2,3})
3x3 ContMechTensors.SymmetricTensor{2,3,Float64,6}:
 0.251845   0.41956   0.0722492
 0.41956    0.411597  0.349139 
 0.0722492  0.349139  0.558384 

julia> σ_m = reshape(σ[:],(3,3))
3x3 Array{Float64,2}:
 0.251845   0.41956   0.0722492
 0.41956    0.411597  0.349139 
 0.0722492  0.349139  0.558384 

julia> λ,ϕ = eig(σ_m)
([-0.1425434358409472,0.3660711635160214,0.9982992898287092],
3x3 Array{Float64,2}:
  0.677725  -0.593174  -0.434551
 -0.683679  -0.29077   -0.669355
  0.270689   0.750731  -0.602602)

julia> Λ = convert(Tensor{2,3},diagm(λ))
3x3 ContMechTensors.Tensor{2,3,Float64,9}:
 -0.142543  0.0       0.0     
  0.0       0.366071  0.0     
  0.0       0.0       0.998299

julia> Φ = convert(Tensor{2,3},ϕ)
3x3 ContMechTensors.Tensor{2,3,Float64,9}:
  0.677725  -0.593174  -0.434551
 -0.683679  -0.29077   -0.669355
  0.270689   0.750731  -0.602602

julia> σ2 = Φ*Λ*Φ'
3x3 ContMechTensors.Tensor{2,3,Float64,9}:
 0.251845   0.41956   0.0722492
 0.41956    0.411597  0.349139 
 0.0722492  0.349139  0.558384

julia> norm(σ-σ2)
1.2866737660321158e-15

```

Thoughts?

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes #14 

``` jl
julia> σ = rand(SymmetricTensor{2,3})
3x3 ContMechTensors.SymmetricTensor{2,3,Float64,6}:
 0.435114    0.916302  0.00930137
 0.916302    0.513069  0.834022  
 0.00930137  0.834022  0.459121  

julia> Λ, Φ = eig(σ)
([-0.75528,0.438991,1.72359],
3x3 ContMechTensors.Tensor{2,3,Float64,9}:
  0.534159   0.672629   -0.512098
 -0.698771   0.0103567  -0.715271
  0.475808  -0.739907   -0.475546)
```

CC @fredrikekre

### CODE REVIEW

**Comment by KristofferC** [comment]
Samt:

``` jl
julia> diagm(SymmetricTensor{2,3}, Λ)
3x3 ContMechTensors.SymmetricTensor{2,3,Float64,6}:
 -0.75528  0.0       0.0    
  0.0      0.438991  0.0    
  0.0      0.0       1.72359

julia> diagm(Tensor{2,3}, Λ)
3x3 ContMechTensors.Tensor{2,3,Float64,9}:
 -0.75528  0.0       0.0    
  0.0      0.438991  0.0    
  0.0      0.0       1.72359
```

File: src/symmetric_ops.jl [hunk 1]
```diff
@@ -174,3 +174,14 @@ end
         SymmetricTensor{4, dim, $Tv, $N}(A_otimes_B(S1.data, S2.data))
     end
 end
+
+#######
+# Dev #
```
Comment (fredrikekre) [inline src/symmetric_ops.jl]: eig

**Comment by fredrikekre** [comment]
Sweet!
Sorry to ruin your lunch :p

**Comment by KristofferC** [comment]
Already eaten ;)

**Pull request merged**